src/buckets/components/context/csvUploaderProvider.tsx

```diff
@@ -0,0 +1,191 @@
+
+        let chunk = ''
+
+        let measurement: any = ''
+        let field: any = ''
+        let time: any = ''
+        let tags: any = ''
+        let line: any = ''
+
+        let counter = 0
+        let progress = 0
+
+        const pendingWrites = []
+        let precision = WritePrecision.Ms
+
+        for (let i = 0; i < length; i++) {
+          if (i !== 0 && i % MAX_CHUNK_SIZE === 0) {
+            // second range
+            if (`${time}`.length === 10) {
+              precision = WritePrecision.S
+            }
+            // millisecond range
+            if (`${time}`.length === 13) {
+              precision = WritePrecision.Ms
+            }
+            // microsecond range
+            if (`${time}`.length === 16) {
+              precision = WritePrecision.Us
+            }
+            // nanosecond range
+            if (`${time}`.length === 19) {
+              precision = WritePrecision.Ns
+            }
+            const resp = postWrite({
+              data: chunk,
+              query: {org: org.name, bucket, precision},
+            }).then(() => {
+              const percent = (++progress / counter) * 100
+              setValue(Math.floor(percent))
+            })
+            pendingWrites.push(resp)
+            counter++
+            chunk = ''
+          }
+          measurement = table.getColumn('_measurement')[i]
+          field = table.getColumn('_field')[i]
+          time = table.getColumn('_time')[i]
+          tags = columns
+            .filter(col => !!table.getColumn(col)[i])
+            .map(col => `${col}=${table.getColumn(col)[i]}`)
+            .join(',')
+            .trim()
+            .replace(/(\r\n|\n|\r)/gm, '')
+          line = `${measurement},${tags} ${field}="${field}" ${time}`
+          chunk = `${line}\n${chunk}`
+        }
+        if (chunk) {
+          // second range
+          if (`${time}`.length === 10) {
+            precision = WritePrecision.S
+          }
+          // millisecond range
+          if (`${time}`.length === 13) {
+            precision = WritePrecision.Ms
+          }
+          // microsecond range
+          if (`${time}`.length === 16) {
+            precision = WritePrecision.Us
+          }
+          // nanosecond range
+          if (`${time}`.length === 19) {
+            precision = WritePrecision.Ns
+          }
+          const resp = postWrite({
+            data: chunk,
+            query: {org: org.name, bucket, precision},
+          }).then(() => {
+            const percent = (++progress / counter) * 100
+            setValue(Math.floor(percent))
+          })
+          pendingWrites.push(resp)
+          counter++
+        }
+        chunk = ''
+        Promise.all(pendingWrites).finally(() => {
+          setUploadFinished(true)
+        })
+      }, 0)
+    },
+    [bucket, org]
+  )
+
+  return (
+    <CsvUploaderContext.Provider
+      value={{
+        handleDismiss,
+        handleDrop,
```

Comment:
drop only makes sense in the context of a view layer that implements drag and drop, where this is more of a `uploadCSV` function